### ISSUE TITLE
multivalue generates illegal wasm code

### ISSUE THREAD

**Issue by L0laapk3** [WebAssembly/wasi-sdk#184]
Hi! I am trying to generate the following code using an external function that has multiple return values: https://godbolt.org/z/WG4nqqsaP.

When I compile it using wasi-sdk-12, it generates an invalid .wasm file. The command I used is:

```
"wasi-sdk/bin/clang++" ^
  -Wl,--no-entry ^
  -nostartfiles ^
  -fno-exceptions ^
  -Wl,--strip-all ^
  -fvisibility=hidden ^
  -mmultivalue ^
  -Xclang -target-abi -Xclang experimental-mv ^
  --sysroot="wasi-sdk/share/wasi-sysroot" ^
  "src/main.cpp" -o "build/bot.wasm"
```

Any idea whats going wrong here?

**Comment by pchickey** [comment]
What tool tells you the resulting wasm is invalid? It may be that the tool doesnt support multivalue, or the support isn't enabled by default.

**Comment by L0laapk3** [comment]
Every tool told me it was invalid, this includes: node 16.3.0 (which has `--experimental-wasm-mv` on), as well as https://webassembly.studio/ and webassembly toolkit plugin for vscode.

A small correction to the original post, with `-Ofast` (or any `-O`) it fails during compilation with an `Fatal: error parsing wasm`. If `-O` is not used, it generates invalid wasm as originally mentioned. (I have used these tools to manually write the expected resulting code using multivalue in .wat and compiled them to .wasm.)

I've attached the invalid wasm as a file, unfortunately github doesn't allow me to directly upload .wasm files so I have added the .txt extension.
[invalid.wasm.txt](https://github.com/WebAssembly/wasi-sdk/files/6657396/invalid.wasm.txt)

The godbolt link in the original post has the source code and expected result.
